src/WebJobs.Extensions.DurableTask.Analyzers/Analyzers/ActivityFunction/ArgumentAnalyzer.cs

```diff
@@ -24,126 +24,117 @@ public class ArgumentAnalyzer
         public static readonly DiagnosticDescriptor InputNotUsedRule = new DiagnosticDescriptor(DiagnosticId, Title, InputNotUsedMessageFormat, Category, Severity, isEnabledByDefault: true, description: Description);
 
         public static void ReportProblems(
-            CompilationAnalysisContext context, 
-            SemanticModel semanticModel, 
-            IEnumerable<ActivityFunctionDefinition> availableFunctions, 
-            IEnumerable<ActivityFunctionCall> calledFunctions)
+            CompilationAnalysisContext context,
+            SemanticModel semanticModel,
+            IEnumerable<ActivityFunctionDefinition> functionDefinitions,
+            IEnumerable<ActivityFunctionCall> functionInvocations)
         {
-            foreach (var activityInvocation in calledFunctions)
+            foreach (var invocation in functionInvocations)
             {
-                var functionDefinition = availableFunctions.Where(x => x.FunctionName == activityInvocation.Name).FirstOrDefault();
-                if (functionDefinition != null)
+                var definition = functionDefinitions.Where(x => x.FunctionName == invocation.FunctionName).FirstOrDefault();
```

Comment:
Find is only implemented for List, but I could change these to Lists.

Reply:
I didn't actually realize that Find() wasn't a linq method. In fact, after reading [this SO answer](https://stackoverflow.com/a/9335036/9035640), I think I prefer the Linq() approach. I would stick with what you have, unless you want to use the overload of .FirstOrDefault() that takes a predicate instead of chaining a .Where().FirstOrDefault()